### ISSUE TITLE
Changes of ESP8266/ESP32 boards.

### ISSUE THREAD

**Issue by gshubham586** [infernoDison/cottonCandy#6]
Hi @infernoDison ,
Hope you are doing great, I want to know What changes should I do so that codes (gateway.ino and node.ino) may compatible with ESP8266 or ESP32 board.

**Comment by infernoDison** [comment]
Hi, 

A disclaimer: I only used ESP32 boards a few times in the past for some Wi-Fi projects, so I am not an ESP32 expert. We have not tested our software on ESP boards.

I don't think you need to change much in the existing gateway.ino and node.ino code except the following

1. I assume that you are probably using the on-board LoRa transceiver of the ESP board, so the first thing is to write your own device driver for sending and receiving messages. Since there might be some libraries for that, you essentially need to build a wrapper. AdafruitDeviceDriver might be a good example.

2. Cottoncandy does not use any additional GPIO pins, besides the ones for connecting to an external transceiver (you probably have noticed them as macros defined in the sketch file). Depending on the transceiver you are using, you might need to handle those pin mappings (can simply do it in the device driver).

A more recent version of Cottoncandy is in the "power" branch, which uses a bare-bones circuit and a real-time clock (RTC) to save as much power as possible. This version currently uses many features that are specific to Atmega328P, so I do not recommend using this version.

**Comment by gshubham586** [comment]
I am using the Lora Ra-02 module which is connected to ESP32 using SPI pins. Can you please tell me where can I change the frequency in the code because I want to use it on 433MHz?

**Comment by infernoDison** [comment]
If you have the code for frequency setup using SPI, you can simply insert it at the `void setup()` in the Arduino sketch (or whatever before `manager->run()` is called in `void loop()`).

Also, in your own driver class (say "LoRa_Ra_02_Driver") which should inherit from the generic "DeviceDriver", you can override the `init()` function to initialize your device frequency there. This might be more elegant.

Please note that you need to provide your own code for the particular device. Cottoncandy currently does not have a built-in software driver for LoRa Ra-02 module. Different transceivers usually come with different interfaces, so we designed Cottoncandy as a generic mesh networking library which is independent of the underlying hardware.

**Comment by jwktje** [comment]
I also want to test this library on an ESP32 board. If anything comes up I might create a PR to make it easier for other people to use with ESP32. Quick question, does the library use some form of encryption?

**Comment by infernoDison** [comment]
@jwktje No, the library does not have encryption right now. In the future, we might add encryptions using the Arduino crypto library, but we need to evaluate its overhead and impact on the network performance.

Note that the EByte transceivers have built-in encryptions (but it is a blackbox to us).
